src/devices/RpiCpuTemperature/RpiCpuTemperature.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Iot.Device.Rpi
+{
+    public class CpuTemperature
+    {
+        public double? ReadTemperature()
+        {
+            double? _temperature = null;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) && File.Exists("/sys/class/thermal/thermal_zone0/temp"))
+            {
+                using (FileStream _fileStream = new FileStream("/sys/class/thermal/thermal_zone0/temp", FileMode.Open, FileAccess.Read))
+                using (StreamReader _reader = new StreamReader(_fileStream))

```

Comment:
nit: StreamReader already disposes stream (there is no harm in double disposing) but you can just construct file stream directly in this call